packages/astro/package.json

```diff
@@ -97,6 +97,8 @@
     "shorthash": "^0.0.2",
     "slash": "^4.0.0",
     "sourcemap-codec": "^1.4.8",
+    "sqlite": "^4.0.23",
+    "sqlite3": "^5.0.2",
```

Comment:
I think this package requires a post-install script to compile the binary, which we want to avoid as a general rule if at all possible. Would love to talk about this more and see if we can get around the sqlite3 requirement.

for example: I only see `WHERE id = X` queries in the PR below. If all you need is an on-disk KV store, Could we just read/write temp files instead? Feels like it could be a perf boost and a usability boost.

Reply:
I could try reading/writing temp files. It will be slower for sure, but it’s worth finding out _how_ much slower.

Reply:
Just out of curiosity, why do we not want post-install scripts to run? Doesn’t esbuild run a post-install script?

Reply:
Doesn't sqlite write out to a file on every update? I assumed temp files would be faster or equal perf, and not slower!

> Just out of curiosity, why do we not want post-install scripts to run? 

You can google around for more in-depth articles on it, but the basics is that it's a huge security vulnerability (most/all "npm security breach" issues are related to post-scripts) and therefore some enterprises disable this. It's good to disable it when you can. It also slows down your install, and prevents a package from working in all environments (ex: I think Skypack disables post-install scripts for security). 

> Doesn’t esbuild run a post-install script?

I know that they've been trying to move away from it via that optional packages trick that makes yarn mad.  We might not be on that latest version yet though.

Reply:
On top of the post-install issue it's also always worth a quick audit on a heavier package or database driver like this: https://npm.anvaka.com/#/view/2d/sqlite3. 

100+ new deps should also be avoided if possible!

Reply:
> Doesn’t sqlite write out to a file on every update? I assumed temp files would be faster or equal perf, and not slower!

Nope—it actually only does one read/write as long as the connection has been open. They claim [35% faster than the file system](https://www.sqlite.org/fasterthanfs.html). That’s a major part of its design.

Reply:
If not on disk, then wouldn't it be storing everything in memory? Or you're just saying it does batch updates when it can? 

Either way, that sounds very neat. TIL!

Reply:
Ah, they go into more detail in that link: 

> So let your take-away be this: read/write latency for SQLite is competitive with read/write latency of individual files on disk. Often SQLite is faster. Sometimes SQLite is almost as fast. Either way, this article disproves the common assumption that a relational database must be slower than direct filesystem I/O.

Reply:
No nothing is stored in memory! It’s 100% file-system based. The PR audit above shows the memory reduction (97% for docs).

Reply:
Cool, this is all great to know.

So just to be clear: I'm still strongly in favor of disk reads/writes over sqlite given the added post-script behavior, the 100+ additional dependency count, and the complexity addition of sqlite over a direct file read/write. The I/O perf win sounds nice but in the context of a full site build I can't imagine it changing things dramatically, maybe more than a few seconds?  Real data could confirm this but I'm just guessing that we're not actually spending that much time doing I/O here, so even a 35% reduction of a few seconds would have a trivial impact.

Also considering that this is a short-term fix to something that will most likely be solved in a larger way by refactoring through our SSR work.

Reply:
That’s fine. This is why it’s a prototype! We should also spin up a fs read/write prototype as well to see the comparison.